UnityProject/Assets/Scripts/Objects/Enterable.cs

```diff
@@ -6,11 +6,41 @@
 namespace Objects
 {
 	/// <summary>
-	/// Allows to trigger an event when a player enter in the space occupied by this component.
+	/// Allows to trigger an event when a player enter the tile of this object.
+	/// Does trigger for ghosts, so check in OnStep
+	/// ServerSide Only
 	/// </summary>
-	public interface IEnterable
+	public interface IPlayerEntersTile
 	{
-		public abstract void OnStep(GameObject eventData);
-		public abstract bool WillStep(GameObject eventData);
+		/// <summary>
+		/// Check whether the player should be allowed to trigger the OnPlayerStep logic
+		/// </summary>
+		/// <param name="playerScript">Playerscript of the player stepping</param>
+		/// <returns>False to block OnPlayerStep, true to run OnPlayerStep</returns>
+		public abstract bool CanPlayerStep(PlayerScript playerScript);
```

Comment:
``CanEnterPlayerStep`` sounds like it should somehow prevent moving into the tile, maybe use something like ``WillAfectPlayer`` following the ``WillInteract`` nomenclature from the interaction framework